File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0;
        int right = numbers.length - 1;
        
        while (left < right) {
            while (numbers[left] + numbers[right] > target)
                right--;
            
            if (numbers[left] + numbers[right] == target)
                return new int[]{left + 1, right + 1};
            else
                left++;
        }
        return null;
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        map = {}
        
        for i in range(len(numbers)):
            if numbers[i] in map.values():
                return [list(map.keys())[list(map.values()).index(numbers[i])] + 1, i + 1]
            else:
                map[i] = target - numbers[i]